frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BaseSlideOutAction.java

```diff
@@ -171,20 +173,23 @@ private void slideOutBranch(AnActionEvent anActionEvent, String branchName) {
     LOG.info("Sliding out '${branchName}' branch in memory");
     val newBranchLayout = branchLayout.slideOut(branchName);
 
-    try {
-      val macheteFilePath = gitRepository.getMacheteFilePath();
-      LOG.info("Writing new branch layout into ${macheteFilePath}");
-      branchLayoutWriter.write(macheteFilePath, newBranchLayout, /* backupOldLayout */ true);
-
-    } catch (BranchLayoutException e) {
-      val exceptionMessage = e.getMessage();
-      val errorMessage = "Error occurred while sliding out '${branchName}' branch" +
-          (exceptionMessage == null ? "" : ": " + exceptionMessage);
-      LOG.error(errorMessage);
-      VcsNotifier.getInstance(project).notifyError(/* displayId */ null,
-          getString("action.GitMachete.BaseSlideOutAction.notification.title.slide-out-fail.HTML").format(branchName),
-          exceptionMessage == null ? "" : exceptionMessage);
-    }
+    ApplicationManager.getApplication()
+        .invokeLater(() -> WriteAction
+            .compute(() -> Try.run(() -> {
```

Comment:
`com.intellij.openapi.application.WriteAction#run(com.intellij.util.ThrowableRunnable<E>)` could be used here too, right?